### ISSUE TITLE
bug: constructor() method no longer runs for components implementing an interface 

### ISSUE THREAD

**Issue by kdeds** [ionic-team/stencil#3773]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.19.0

### Current Behavior

Our components each implement a common interface. We initialize some properties in the constructor(). This used to work fine, but now in Stencil 2.19 the function does not seem to run.

Here is a simple example. In the code below, `this.classProps` is undefined in the render method.
```
import { Component, h} from '@stencil/core';
import { RhclComponent } from "../../model/RhclComponent";

@Component({
  tag: 'rhcl-example',
  styleUrl: 'example.scss',
  shadow: true
})
export class Example implements RhclComponent {

  private classProps: Array<string>;

  constructor() {
    this.classProps = ["variant", "theme"];
  }

  render() {
    console.log(this.classProps); // this is undefined
    return (
      <div>
      </div>
    );
  }
}
```

### Expected Behavior

Code in `constructor()` would run and this.classProps in the above example would be an array with the entries ["variant", "theme"] .

### Steps to Reproduce

1. Create a component which implements an interface.
2. Add some code in the `constructor()`.
3. With Stencil 2.18 installed, verify the code in the constructor runs.
4. With Stencil 2.19 installed, observe that the code in the constructor does not run.

### Code Reproduction URL

https://github.com/kdeds/stencil-constructor-bug

**Comment by kdeds** [comment]
Seems to be related to https://github.com/ionic-team/stencil/commit/bd340b888380929b18e6d56ea98bbe154f19dd77

We moved our constructor code into `componentWillLoad()` which is fine for us but maybe there is something that can be done for backwards compatibility, or documentation to make it clear not to use constructors. This took a while to track down today and was breaking our whole library 😅

**Comment by alicewriteswrongs** [comment]
Ah ok I believe I understand what the problem is. There is an explainer here about the change which resulted in this behavior: https://github.com/ionic-team/stencil/commit/f97783029274f9ee5ea58ba74ab15905c5113c93

In short, to support setting typescript to emit ES2022+ code we need to transpile class fields, like this one:

```ts
class MyComponent {
  @Prop() foo: string = "hey!";
}
```

to the following:

```js
class MyComponent {
  constructor () {
    this.foo = "hey!";
  }
}
```

A bunch of Stencil's runtime behavior for props and so on is based on using `Object.defineProperty` to setup getters and setters, and those don't play so well with the new support for class fields as standardized in ES2022, so we need to do that conversion.

Anywhooo - I looked through our implementation of that transpilation and there was an oversight where existing statements inside of the constructor method are not preserved _in the case where a method has to be added_.

This PR fixes the issue: #3776

**Comment by rwaskiewicz** [comment]
Hey @kdeds 👋 

@alicewriteswrongs' fix has been included in a [pre-release of Stencil, v2.19.2-0](https://github.com/ionic-team/stencil/releases/tag/v2.19.2-0). Can you please install it and verify that it fixes the issue?

**Comment by kdeds** [comment]
Sorry, had some weird issues trying to install it this morning but just gave it another try and got it downloaded. Looks like the fix works! Thank you for the quick turnaround :D
![image](https://user-images.githubusercontent.com/2187441/198413479-a31577e9-7102-445d-8dce-c72bc62cc028.png)
![image](https://user-images.githubusercontent.com/2187441/198413521-3dc2eefe-6ecd-4dab-9552-189f8c5b045d.png)

**Comment by rwaskiewicz** [comment]
The fix for this bug has been released as a part of [Stencil v2.19.2](https://github.com/ionic-team/stencil/releases/tag/v2.19.2). As a result, I'm going to close this issue. Should the bug re-appear, please feel free to open a new issue. Thanks again!
